kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/JdbcAuthenticationProviderImpl.scala

```diff
@@ -0,0 +1,232 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.service.authentication
+
+import java.sql.{Connection, DriverManager, PreparedStatement, Statement}
+import javax.security.sasl.AuthenticationException
+
+import org.apache.commons.lang3.StringUtils
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf._
+
+class JdbcAuthenticationProviderImpl(conf: KyuubiConf) extends PasswdAuthenticationProvider
+  with Logging {
+
+  private val dbDriver = conf.get(AUTHENTICATION_JDBC_DRIVER)
+  private val dbUrl = conf.get(AUTHENTICATION_JDBC_URL)
+  private val dbUserName = conf.get(AUTHENTICATION_JDBC_USERNAME)
+  private val dbPassword = conf.get(AUTHENTICATION_JDBC_PASSWORD)
+  private val querySql = conf.get(AUTHENTICATION_JDBC_QUERY)
+
+  private val SQL_PLACEHOLDER_REGEX = """\$\{.+?}""".r
+  private val USERNAME_SQL_PLACEHOLDER = "${username}"
+  private val PASSWORD_SQL_PLACEHOLDER = "${password}"
+
+  /**
+   * The authenticate method is called by the Kyuubi Server authentication layer
+   * to authenticate users for their requests.
+   * If a user is to be granted, return nothing/throw nothing.
+   * When a user is to be disallowed, throw an appropriate [[AuthenticationException]].
+   *
+   * @param user     The username received over the connection request
+   * @param password The password received over the connection request
+   * @throws AuthenticationException When a user is found to be invalid by the implementation
+   */
+  @throws[AuthenticationException]
+  override def authenticate(user: String, password: String): Unit = {
+    if (StringUtils.isBlank(user)) {
+      throw new AuthenticationException(s"Error validating, user is null" +
+        s" or contains blank space")
+    }
+
+    if (StringUtils.isBlank(password)) {
+      throw new AuthenticationException(s"Error validating, password is null" +
+        s" or contains blank space")
+    }
+
+    checkConfigs
+
+    loadJdbcDriverClass
+
+    var connection: Connection = null
+    var queryStatement: PreparedStatement = null
+
+    try {
+      connection = DriverManager.getConnection(dbUrl.get, dbUserName.orNull, dbPassword.orNull)
+
+      queryStatement = getAndPrepareStatement(connection, user, password)
+
+      val resultSet = queryStatement.executeQuery()
+
+      if (resultSet == null || !resultSet.next()) {
+        // Auth failed
+        throw new AuthenticationException(s"Password does not match or no such user. user:" +
+          s" $user , password length: ${password.length}")
+      }
+
+      // Auth passed
+
+    } catch {
+      case e: AuthenticationException =>
+        throw e
+      case e: Exception =>
+        error("Cannot get user info", e);
+        throw e
+    } finally {
+      closeDbConnection(connection, queryStatement)
+    }
+  }
+
+  private def checkConfigs: Unit = {
```

Comment:
let add `()` if the function has side-effect.

Reply:
thx and added.